### ISSUE TITLE
Openweathermap api will not load location data

### ISSUE THREAD

**Issue by zomboto** [WangDaYeeeeee/GeometricWeather#223]
Just yesterday I received an alert from F-droid stating that the old geometric weather is deprecated and recommending to install the new one. I was excited to see the new app.offers an api other than accuweather but when I choose openweather api it never loads location nor current weather data. The accuweather api works and loads it fast. Is the openweather api not valid? Thank you

**Comment by papjul** [comment]
I can see that account is temporarily blocked due to exceeding requests, wow, didn't know so many people would choose OpenWeatherMap as their provider.

@WangDaYeeeeee Could you please contact OpenWeatherMap to lift the limit?

> If you exceed the 1 million calls, you can request additional calls for free as per the FAQ https://openweathermap.org/faq
> 
> Question "API calls return an error 401" on bottom:
> > **For FOSS developers:** we welcome free and open source software and are willing to help you. If you want to use OpenWeather data in your free software application please register an API key and file a ticket describing your application and API key registered. OpenWeather will review your request lift access limits for your key if used in open source application.

I also think that developing a per-API configuration menu will become urgent, so people can choose to provide their own key(s).

**Comment by zomboto** [comment]
Ok thank you @papjul , I'm inexperienced in the area of API keys so I'm glad to know it wasn't something I was doing wrong. Thank you and everyone for your hard work!

**Comment by legacychimera247** [comment]
Out of curiosity, any news on openweather?because accuweather is one of the most inaccurate despite it's name...

**Comment by papjul** [comment]
Yes, you can input your own API key if you build from sources. This will workaround this issue.

**Comment by legacychimera247** [comment]
Unfortunately i'm not too skilled to do that, but i read that it will be a feature for next update, so i will wait for that...

**Comment by eoli3n** [comment]
any news ? it would be great to be able to use OpenWeather

**Comment by papjul** [comment]
> any news ? it would be great to be able to use OpenWeather

You need to register your own API key on OpenWeatherMap and input it in latest master build of Geometric Weather.

**Comment by eoli3n** [comment]
Is it released to f-droid ?

**Comment by papjul** [comment]
> When will it be released to f-droid ?

Only @WangDaYeeeeee can make a release tag ;)

**Comment by mu88** [comment]
Any news when we can celebrate the next release? :)

**Comment by zetaomegagon** [comment]
@WangDaYeeeeee can you add a release tag for current master, based on the below message?

> > any news ? it would be great to be able to use OpenWeather
> 
> You need to register your own API key on OpenWeatherMap and input it in latest master build of Geometric Weather.

TYVM!

**Comment by papjul** [comment]
The release containing the new feature is available from F-Droid or [releases page](https://github.com/WangDaYeeeeee/GeometricWeather/releases). V3.100 or later.

By the way, #368 is a future fix that will show a toast message explaining why data cannot be loaded.

**Comment by camoz** [comment]
Thank you! :)

**Comment by zetaomegagon** [comment]
@papjul thanks for clarifying.

In case someone else gets confused, the setting you want is `Settings > Service provider > Advanced service provider settings`

**Comment by EmberHeartshine** [comment]
Using your own API key also does not work here. Same error, even on a freshly-generated key.
